diff --git a/build/config/ui.gni b/build/config/ui.gni
index 55478b02cd0fe..42e2205845fe5 100644
--- a/build/config/ui.gni
+++ b/build/config/ui.gni
@@ -37,8 +37,7 @@ declare_args() {
 
 assert(!use_glib || (is_linux && !is_chromecast))
 
-use_atk = is_linux && !is_chromecast && use_glib &&
-          current_toolchain == default_toolchain
+use_atk = false
 
 # Whether using Xvfb to provide a display server for a test might be
 # necessary.
