diff --git a/third_party/widevine/cdm/widevine.gni b/third_party/widevine/cdm/widevine.gni
index 1a833ae57589c..b50ee8889628a 100644
--- a/third_party/widevine/cdm/widevine.gni
+++ b/third_party/widevine/cdm/widevine.gni
@@ -25,7 +25,7 @@ if (is_chromeos && !is_chromeos_device) {
 # supported via Android MediaDrm API.
 library_widevine_cdm_available =
     (is_chromeos_ash && (target_cpu == "x64" || target_cpu == "arm")) ||
-    ((target_os == "linux" || is_chromeos_lacros) &&
+    ((target_os == "linux" || is_chromeos_lacros || is_ironos) &&
      (target_cpu == "x86" || target_cpu == "x64")) ||
     (target_os == "mac" && (target_cpu == "x64" || target_cpu == "arm64")) ||
     (target_os == "win" && (target_cpu == "x86" || target_cpu == "x64"))
diff --git a/tools/variations/fieldtrial_to_struct.py b/tools/variations/fieldtrial_to_struct.py
index 9ecf38974795f..3afde792b94d5 100755
--- a/tools/variations/fieldtrial_to_struct.py
+++ b/tools/variations/fieldtrial_to_struct.py
@@ -41,6 +41,7 @@ _platforms = [
     'linux',
     'mac',
     'windows',
+    'ironos',
 ]
 
 _form_factors = [
@@ -217,6 +218,9 @@ def main(arguments):
   if not opts.platform:
     parser.error('You must specify at least 1 --platform.')
 
+  if opts.platform == 'ironos':
+      opts.platform = 'linux'
+
   description_filename = os.path.normpath(args[0])
   shortroot = opts.output
   if opts.destdir:
