diff --git a/package/gcc/gcc-final/gcc-final.mk b/package/gcc/gcc-final/gcc-final.mk
index 99d7047b5f..dde2406c34 100644
--- a/package/gcc/gcc-final/gcc-final.mk
+++ b/package/gcc/gcc-final/gcc-final.mk
@@ -165,6 +165,10 @@ define GCC_FINAL_INSTALL_STAGING_CMDS
 		cp -dpf $(HOST_GCC_FINAL_GCC_LIB_DIR)/$(lib)* \
 			$(STAGING_DIR)/lib/
 	)
+	-cp -dpf $(HOST_DIR)/lib*/gcc/$(GNU_TARGET_NAME)/$(GCC_FINAL_VERSION)/libgcc.a \
+			$(STAGING_DIR)/lib/
+	-cp -dpf $(HOST_DIR)/lib*/gcc/$(GNU_TARGET_NAME)/$(GCC_FINAL_VERSION)/*.o \
+			$(STAGING_DIR)/lib/
 	$(foreach lib,$(GCC_FINAL_USR_LIBS), \
 		cp -dpf $(HOST_GCC_FINAL_GCC_LIB_DIR)/$(lib)* \
 			$(STAGING_DIR)/usr/lib/
