diff --git a/package/weston/weston.hash b/package/weston/weston.hash
index 35fe608e20..a213764619 100644
--- a/package/weston/weston.hash
+++ b/package/weston/weston.hash
@@ -1,4 +1,2 @@
 # From https://lists.freedesktop.org/archives/wayland-devel/2025-April/044158.html
-sha256  b47216b3530da76d02a3a1acbf1846a9cd41d24caa86448f9c46f78f20b6e0ac  weston-14.0.2.tar.xz
-sha512  e8214ec893e6c3ae94eb3c92feba104b0201843e9143f726a3e9a4d396d02523c94da706c1348cf934bc339fb1a4bc1fecdb865f0ea914115fd346d9eda091f5  weston-14.0.2.tar.xz
-sha256  fdb65868f65d0fbdb05c2d3b779e10ce9969fa0c4b9262ba4f260e87086ab860  COPYING
+sha256  7d0403fcbff5dded289ff21fb86b8dbbd3074c999d47ee27dc6f5c32df790ac2  kiosk-sleep-v14.0.3.tar.gz
diff --git a/package/weston/weston.mk b/package/weston/weston.mk
index 7d43132073..ab9896fa0e 100644
--- a/package/weston/weston.mk
+++ b/package/weston/weston.mk
@@ -4,9 +4,9 @@
 #
 ################################################################################
 
-WESTON_VERSION = 14.0.2
-WESTON_SITE = https://gitlab.freedesktop.org/wayland/weston/-/releases/$(WESTON_VERSION)/downloads
-WESTON_SOURCE = weston-$(WESTON_VERSION).tar.xz
+WESTON_VERSION = 14.0.3
+WESTON_SITE = https://github.com/iron-os/weston/archive/refs/heads
+WESTON_SOURCE = kiosk-sleep-v$(WESTON_VERSION).tar.gz
 WESTON_LICENSE = MIT
 WESTON_LICENSE_FILES = COPYING
 WESTON_CPE_ID_VENDOR = wayland
