diff --git a/package/weston/weston.hash b/package/weston/weston.hash
index 35fe608e20..a213764619 100644
--- a/package/weston/weston.hash
+++ b/package/weston/weston.hash
@@ -1,4 +1,2 @@
 # From https://lists.freedesktop.org/archives/wayland-devel/2024-October/043858.html
-sha256  a8150505b126a59df781fe8c30c8e6f87da7013e179039eb844a5bbbcc7c79b3  weston-14.0.1.tar.xz
-sha512  2d782f98f28b39b990acecfc885f62ab34c9cdd22ef06b09c5d8df792112c47b365f89c2bf05d871e36ddfb4c49aa7e6d05ff60fe67a2657727162e4f564a412  weston-14.0.1.tar.xz
-sha256  fdb65868f65d0fbdb05c2d3b779e10ce9969fa0c4b9262ba4f260e87086ab860  COPYING
+sha256  7acab95557f2f91f94fba918acbfdd916e58d597b4b20af9664be7ec36b954eb  kiosk-sleep-v14.0.2.tar.gz
diff --git a/package/weston/weston.mk b/package/weston/weston.mk
index 7d43132073..ab9896fa0e 100644
--- a/package/weston/weston.mk
+++ b/package/weston/weston.mk
@@ -4,9 +4,9 @@
 #
 ################################################################################
 
-WESTON_VERSION = 14.0.1
-WESTON_SITE = https://gitlab.freedesktop.org/wayland/weston/-/releases/$(WESTON_VERSION)/downloads
-WESTON_SOURCE = weston-$(WESTON_VERSION).tar.xz
+WESTON_VERSION = 14.0.2
+WESTON_SITE = https://github.com/iron-os/weston/archive/refs/heads
+WESTON_SOURCE = kiosk-sleep-v$(WESTON_VERSION).tar.gz
 WESTON_LICENSE = MIT
 WESTON_LICENSE_FILES = COPYING
 WESTON_CPE_ID_VENDOR = wayland
