diff --git a/package/weston/weston.hash b/package/weston/weston.hash
index b78a7c7550..8b5c53c261 100644
--- a/package/weston/weston.hash
+++ b/package/weston/weston.hash
@@ -1,6 +1 @@
-# From https://lists.freedesktop.org/archives/wayland-devel/2022-February/042103.html
-md5  bc4abe2ee6904a4890a0c641c6257f91  weston-10.0.0.tar.xz
-sha1  b1a51e3bb7ee1d75bbf728c8071771e07fc97fdd  weston-10.0.0.tar.xz
-sha256  5c23964112b90238bed39e5dd1e41cd71a79398813cdc3bbb15a9fdc94e547ae  weston-10.0.0.tar.xz
-sha512  77e05530751e4f743eebf015ab5b78a1fcc66b774713b5e296cfffc488be84e459f811c85ac4280b7b63eebf4ff5c958dd7306376ab343e6b80feae16c8c35a4  weston-10.0.0.tar.xz
-sha256  fdb65868f65d0fbdb05c2d3b779e10ce9969fa0c4b9262ba4f260e87086ab860  COPYING
+sha256  a05c79256b639abcc6efaaf3a286b7f48096a7c9abcd7475f479f024f1e8eeac  kiosk-sleep-v10.0.0.tar.gz
diff --git a/package/weston/weston.mk b/package/weston/weston.mk
index 90cf160688..699251ebd6 100644
--- a/package/weston/weston.mk
+++ b/package/weston/weston.mk
@@ -5,8 +5,8 @@
 ################################################################################
 
 WESTON_VERSION = 10.0.0
-WESTON_SITE = https://wayland.freedesktop.org/releases
-WESTON_SOURCE = weston-$(WESTON_VERSION).tar.xz
+WESTON_SITE = https://repos.lvgd.ch/iron-os/weston/archive
+WESTON_SOURCE = kiosk-sleep-v$(WESTON_VERSION).tar.gz
 WESTON_LICENSE = MIT
 WESTON_LICENSE_FILES = COPYING
 WESTON_CPE_ID_VENDOR = wayland
